src/quo2/components/drawers/documentation_drawers/view.cljs

```diff
@@ -26,15 +26,17 @@
       :style               {:color (colors/theme-colors colors/neutral-100
                                                         colors/white
                                                         (when shell? :dark))}
-      :weight              :semi-bold} title]
+      :weight              :semi-bold}
+     title]
     [rn/view {:style style/content :accessibility-label :documentation-drawer-content}
      content]
     (when show-button?
       [button/button
-       (merge {:size                24
-               :type                (if shell? :blur-bg-outline :outline)
-               :on-press            on-press-button
-               :accessibility-label :documentation-drawer-button
-               :after               button-icon}
-              (when shell? {:override-theme :dark})) button-label])]])
+       (cond-> {:size                24
+                :type                (if shell? :blur-bg-outline :outline)
+                :on-press            on-press-button
+                :accessibility-label :documentation-drawer-button
+                :after               button-icon}
+         shell? (assoc :override-theme :dark))
+       button-label])]])
```

Comment:
Since merge is more expensive and we just want to add a new key to the map, I refactored to use a `cond->` and `assoc` instead of  `merge` + `when`